### ISSUE TITLE
Error start warranty date DELL

### ISSUE THREAD

**Issue by stephanehofman** [InfotelGLPI/manufacturersimports#106]
Dear GLPi user.

BEFORE SUBMITTING YOUR ISSUE, please make sure to read and follow these steps :

* Verify that your question has not already been asked
* Please use the below template.
* Delete this text before submiting your issue.

The Plugin team.

------------
* Version of the plugin : 
2.3.2

* Version of your GLPI : 
9.5.7

* Steps to reproduce (which actions have you made) :
When I want import Dell Warranty informations, the import works. But the start warranty date is wrong
Buy date is Ok.
End of warranty is OK
Duration has an error
Start date has an error too 

* Expected result :
Start warranty date is identical to buy date

* Actual result :
start warranty date is wrong

* URL of the page :

* Screenshot of the problem (if pertinent) :
![Error date](https://user-images.githubusercontent.com/3272820/158989458-07d8c502-ddc0-445f-aaf9-fcfeab585cde.png)

**Comment by BLINGjj** [comment]
I think i saw the same issue as u. 
I'll try to explain. 
On ur example , the duration of warranty is 36 month right?

**Comment by stephanehofman** [comment]
yes ...

**Comment by BLINGjj** [comment]
I have same on my warranty , i don't fix it's bug , that add 1 year at u start warranty + 24 month ( depend on warranty i guess i'll check this tomorrow )

**Comment by stephanehofman** [comment]
This the same problem with 2, 3 4 or 5 years...
The bug for me is that the start date must be the same as the date of purchase, no matter the lenght of the warranty.

**Comment by cadegenn** [comment]
Hi,
Same problem here with GLPI 10.0.5 and plugin ManufacturerImport 3.0.5
Plugin return erroneous informations with start warranty date in the future : 
![Sélection_158](https://user-images.githubusercontent.com/6342490/202730050-662498d9-e4dd-4f7f-918b-6fdc8ea0d847.png)
the data at Dell URL is as follow : 
![Sélection_159](https://user-images.githubusercontent.com/6342490/202730139-1c134a6b-8b4d-4ef6-9fbd-4b630c62ce15.png)
Maybe a confusion with different warranty levels ?

More of a concern, plugin seems to have updated old computers with fancy warranty dates : we got old GX620 (Dell computer from early 2010' ? can't remember exactly) with a beginning warranty set to 2022 :open_mouth: 
![Sélection_160](https://user-images.githubusercontent.com/6342490/202731444-668710b4-ffa1-4d99-918c-0c74a5720d58.png)
but I guess it is another problem.

**Comment by tsmr** [comment]
Can you give me the serial used ?

**Comment by cadegenn** [comment]
sure, can I (should I) send this via private message ?

**Comment by tsmr** [comment]
thetsmr[at]gmail.com

**Comment by cadegenn** [comment]
Hi,
I played a lot with Dell warranties using powershell. What I found is that Dell API return empty warranty dates for (very) old PC.
If think that if date is null, plugin ManufacturersImport use current date. As such, all our old PC got their warranty start date replaced by the current date plugin was last run.

I found that the Dell API return correct ShipDate. Is it possible to use this date as a reference in case other dates are empty ?

Thank you.
